pkg/counter/counter.go

```diff
@@ -12,73 +12,74 @@ type Counter struct {
 	value uint64
 }
 
-// NewCounter creates a new counter with the given initial value.
 func NewCounter(initialValue uint64) Counter {
 	return Counter{value: initialValue}
 }
 
-// Increase increases the counter by the given value (thread-safe).
-func (c *Counter) Increase(values ...uint64) error {
-	_, err := c.IncreaseAndRead(values...)
-	return err
-}
+// Increment
 
-// IncreaseAndRead increases the counter by the given value (thread-safe) and returns the new value.
-func (c *Counter) IncreaseAndRead(values ...uint64) (uint64, error) {
-	if len(values) == 0 {
-		values = append(values, 1)
+// Increment increments the counter by given value (default: 1, thread-safe).
+func (c *Counter) Increment(x ...uint64) error {
+	val := uint64(1)
+	if len(x) != 0 {
+		for _, v := range x {
+			val += v
+		}
+		val--
```

Comment:
`val++`?

Reply:
Oh, never mind.

Reply:
Could you elaborate ?

Reply:
I wrongly thought that was a typo. But noticed that it was removing the initial 1.

Reply:
Might be nice to add a comment here, but not critical.